dataprofiler/profilers/profile_builder.py

```diff
@@ -1857,6 +1864,47 @@ def _merge_correlation_helper(corr_mat1, mean1, std1, n1,
 
         return corr_mat
 
+    def _update_chi2(self):
+        n_cols = len(self._profile)
+        cat_cols = {}
+        cat_col_idxs = []
+
+        # Get categorical columns
+        for i in range(n_cols):
+            compiler = self._profile[i]
+            data_stats_compiler = compiler.profiles["data_stats_profile"]
+            profiler = data_stats_compiler._profiles["category"]
+            if profiler.is_match:
+                categories = profiler._categories
+                sample_size = profiler.sample_size
+                cat_cols[i] = (categories, sample_size)
+                cat_col_idxs.append(i)
+
+        # Fill matrix with nan initially
+        chi2_mat = np.full((n_cols, n_cols), np.nan)
+        # Compute chi_sq for each
+        for i in range(len(cat_col_idxs)):
+            for j in range(i, len(cat_col_idxs)):
+                if i == j:
+                    chi2_mat[i][j] = 1
+                    continue
+                idx1 = cat_col_idxs[i]
+                idx2 = cat_col_idxs[j]
+
+                categories1 = cat_cols[idx1][0]
+                categories2 = cat_cols[idx2][0]
+                sample_size1 = cat_cols[idx1][1]
+                sample_size2 = cat_cols[idx2][1]
```

Comment:
these 2 can be done in a single line via:
`categories1, sample_size1 = cat_cols[idx1]`
likewise for the set 2